### PR TITLE
FreeBSD preliminary work.

### PR DESCRIPTION
**Pull request by nicolasnoble**
-) You can't assume bash is installed. Scripts needs to be cleaned out of bashisms.
-) You can't assume python is in /usr/bin. Use env instead.
-) AF_INET is in sys/socket.h
-) Added port_platform's basic structure for FreeBSD, based off Darwin.
-) FreeBSD doesn't have and doesn't need libdl for OpenSSL.

### CODE REVIEW

**Comment by nicolasnoble** [comment]
I haven't checked if this breaks MacOS yet. Maybe it does. Maybe.

**Comment by nicolasnoble** [comment]
Also, this is a basic fix for #1035. The libraries will compile, and some of the tests will work.

File: test/core/end2end/gen_build_json.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.7
```
Comment (dklempner) [inline test/core/end2end/gen_build_json.py]: For what it's worth, I'm told that the weird python=python3 with python2 installed configuration is against recommendation, so dropping the explicit selection of python2 here as you are doing is probably fine.

**Comment by nicolasnoble** [comment]
I've updated that PR so it becomes mergeable. I'll assign @vjpai to it. Maybe @M-Vivek can try it out and confirm ?

**Comment by vjpai** [comment]
The code changes here look good, but I want to know what the status is with regards to the tests. Do all the tests pass for C, C++, other languages?

**Comment by nicolasnoble** [comment]
No, the tests won't even compile at the moment - hence the world "preliminary" up there :-)

**Comment by vjpai** [comment]
Ok, I get it. Sure, then, let's get this in so that we can make further progress.

**Comment by M-Vivek** [comment]
had a chance to run the C/C++ test cases on freeBSD10. Out of around 400
odd cases for C, around 8 failed. And in C++, some 2 test cases failed.

And to run the test cases I had to do some Makefile changes and other
stuffs which I am recollecting, which i will post later. However, this
shouldn't be a concern I believe.

So, We are good to go here for freeBSD10!

**Pull request merged**